apps/portal/app/routes/app+/list+/$id.tsx

```diff
@@ -13,73 +19,98 @@ import ImageModal from '@components/profile/image-modal'
 import ShareCta from '@components/share-cta'
 import ShareModal from '@components/share-modal'
 import { useGoBack } from '@lib/hooks/useGoBack'
-import { useLiveLoader } from '@lib/hooks/useLiveLoader'
 import {
   addIdentitiesListModalAtom,
   imageModalAtom,
   shareModalAtom,
 } from '@lib/state/store'
-import logger from '@lib/utils/logger'
 import { invariant } from '@lib/utils/misc'
 import { json, LoaderFunctionArgs } from '@remix-run/node'
-import { Outlet, useNavigate } from '@remix-run/react'
-import { fetchWrapper } from '@server/api'
+import { Outlet, useLoaderData, useNavigate } from '@remix-run/react'
 import { requireUser, requireUserWallet } from '@server/auth'
-import { getVaultDetails } from '@server/multivault'
+import { QueryClient } from '@tanstack/react-query'
 import {
   BLOCK_EXPLORER_URL,
   IPFS_GATEWAY_URL,
+  MULTIVAULT_CONTRACT_ADDRESS,
   NO_PARAM_ID_ERROR,
   NO_WALLET_ERROR,
   PATHS,
 } from 'app/consts'
 import TwoPanelLayout from 'app/layouts/two-panel-layout'
-import { VaultDetailsType } from 'app/types'
 import { useAtom } from 'jotai'
 
 export async function loader({ request, params }: LoaderFunctionArgs) {
   const user = await requireUser(request)
   invariant(user, 'User not found')
   invariant(user.wallet?.address, 'User wallet not found')
 
+  const queryClient = new QueryClient()
+
   const userWallet = await requireUserWallet(request)
   invariant(userWallet, NO_WALLET_ERROR)
   const id = params.id
   invariant(id, NO_PARAM_ID_ERROR)
 
-  const claim = await fetchWrapper(request, {
-    method: ClaimsService.getClaimById,
-    args: { id },
+  const [, objectId] = id.split('-')
+  invariant(objectId, 'Object ID not found in composite ID')
+
+  // const listsLimit = parseInt(url.searchParams.get('effectiveLimit') || '200')
+  // const listsOffset = parseInt(url.searchParams.get('listsOffset') || '0')
+  // const listsOrderBy = url.searchParams.get('listsSortBy')
+
+  const tripleResult = await fetcher<GetTripleQuery, GetTripleQueryVariables>(
+    GetTripleDocument,
+    {
+      tripleId: objectId,
+    },
+  )()
+
+  await queryClient.prefetchQuery({
+    queryKey: ['get-triple', { id: objectId }],
+    queryFn: () => tripleResult,
   })
 
-  let vaultDetails: VaultDetailsType | null = null
-
-  if (claim !== undefined && claim.vault_id) {
-    try {
-      vaultDetails = await getVaultDetails(
-        claim.contract,
-        claim.vault_id,
-        userWallet as `0x${string}`,
-      )
-    } catch (error) {
-      logger('Failed to fetch vaultDetails', error)
-      vaultDetails = null
-    }
-  }
+  const objectResult = await fetcher<GetAtomQuery, GetAtomQueryVariables>(
+    GetAtomDocument,
+    {
+      id: objectId,
+    },
+  )()
+
+  await queryClient.prefetchQuery({
+    queryKey: ['get-object', { id: objectId }],
+    queryFn: () => objectResult,
+  })
 
   return json({
```

Comment:
Same as in the other route -- be sure to include `dehydratedState: dehydrate(queryClient),` in the `json` response -- this'll tap into the `<HydrationBoundary/>` we have in the provider. No need to add this in the client loader data since it's at the provider level. If the query keys match and the dehydratedState is passed, the client queries (should) pick it up